File: 프로그래머스/0/181860. 빈 배열에 추가， 삭제하기/빈 배열에 추가， 삭제하기.js
function solution(arr, flag) {
    return flag.reduce((acc, cur, idx) => {
        if(cur) {
            acc.push(...Array(arr[idx] * 2).fill(arr[idx]));        
        } else {
            acc.splice(-arr[idx]);
        }
        return acc;
    }, [])
}

File: 프로그래머스/0/181860. 빈 배열에 추가， 삭제하기/빈 배열에 추가， 삭제하기.java
import java.util.*;

class Solution {
    public int[] solution(int[] arr, boolean[] flag) {
        List<Integer> result = new ArrayList<>();
        
        for (int i = 0; i < arr.length; i++) {
            int j = 0;
            
            if (flag[i]) {
                while (j < arr[i] * 2) {
                    result.add(arr[i]);
                    j++;
                }
            } else {
                while (j < arr[i]) {
                    result.remove(result.size() - 1);
                    j++;
                }
            }
        }
        
        return result.stream().mapToInt(Integer::intValue).toArray();
    }
}